accord-core/src/test/java/accord/impl/MessageListener.java

```diff
@@ -117,6 +127,18 @@ public void onClientAction(ClientAction action, Node.Id from, TxnId id, Object m
             }
         }
 
+        private Topology previous = null;
+
+        @Override
+        public void onTopologyChange(Topology topology)
+        {
+            if (previous != null)
+            {
+                logger.debug("Topology Change {} -> {}: {}", previous.epoch(), topology.epoch(), TopologyUtils.diff(previous, topology));
```

Comment:
honestly I can drop the diff... I mostly did it to confirm where things happened but personally found `ListStore.history` easier as it shows the local history (including bootstrap) for a single key... I did find the epoch notification helpful though to know when this changes